.github/workflows/ci.yaml

```diff
@@ -0,0 +1,24 @@
+name: "ci"
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
```

Comment:
Lets use the 3.11 version declared in `pyproject.toml`

Reply:
good call, I'll update this